Intersect.Server/Entities/Player.cs

```diff
@@ -3609,6 +3609,25 @@ public void CraftItem(Guid id)
         {
             if (CraftingTableId != Guid.Empty)
             {
+                var table = CraftingTableBase.Get(CraftingTableId);
+                if (table == null)
+                    return;
+
+                var craftToCheck = CraftBase.Get(id);
+
+                if (craftToCheck == null)
+                    return;
+
+                if (!table.Crafts.Contains(craftToCheck.Id))
```

Comment:
Isn't `craftToCheck.Id` equal to `id`? If so, this check can be moved to _before_ `CraftBase.Get(id)` and avoid an expensive lookup call.